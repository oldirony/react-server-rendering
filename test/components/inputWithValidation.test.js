import React from 'react';
import InputWithValidation from '../../src/components/input-with-validation/input-with-validation';

const props = {
	meta: {
		touched: true
	},
	placeholder: "placeholder"
}
test('Base rendering', ()=>{
	let input = shallow(<InputWithValidation {...props}/>);
	expect(input).toMatchSnapshot();

	input = shallow(<InputWithValidation {...props} meta={
		{...props.meta, error: 'Big error'}
	}/>);

	expect(input).toMatchSnapshot();
});