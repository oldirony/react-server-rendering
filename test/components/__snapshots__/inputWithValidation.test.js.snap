// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base rendering 1`] = `
<div
  className="field-wrapper"
>
  <input
    className=""
    placeholder="placeholder"
  />
</div>
`;

exports[`Base rendering 2`] = `
<div
  className="field-wrapper"
>
  <input
    className="isInvalid"
    placeholder="placeholder"
  />
  <div
    className="errorMessage"
  >
    Big error
  </div>
</div>
`;
